javatests/arcs/android/storage/service/StorageServiceNgImplTest.kt

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.storage.service
+
+import android.os.IBinder
+import android.os.Parcel
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.android.crdt.ParcelableCrdtType
+import arcs.android.storage.writeStoreOptions
+import arcs.core.data.CountType
+import arcs.core.storage.FixedDriverFactory
+import arcs.core.storage.StorageKey
+import arcs.core.storage.StorageKeyParser
+import arcs.core.storage.StoreOptions
+import arcs.core.storage.UntypedProxyMessage
+import arcs.core.storage.keys.RamDiskStorageKey
+import arcs.core.storage.testutil.MockDriverProvider
+import arcs.core.storage.testutil.testWriteBackProvider
+import com.google.common.truth.Truth.assertThat
+import kotlin.test.assertFailsWith
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class StorageServiceNgImplTest {
+
+  private val driverFactory = FixedDriverFactory(MockDriverProvider())
+  private val onProxyMessageCallback: suspend (StorageKey, UntypedProxyMessage) -> Unit =
+    { _: StorageKey, _: UntypedProxyMessage -> }
+  private val storageKey = RamDiskStorageKey("myCount")
+  private val messageCallback = object : IMessageCallback {
+    override fun onMessage(encodedMessage: ByteArray?) {}
+
+    override fun asBinder(): IBinder? { return null }
+  }
+  private val channelCallback = object : IStorageChannelCallback {
+    override fun onCreate(channel: IStorageChannel) {}
+
+    override fun asBinder(): IBinder? { return null }
+  }
+
+  @Before
+  fun setUp() {
+    StorageKeyParser.addParser(RamDiskStorageKey)
+  }
+
+  @Test
+  fun opensStorageChannel() = runBlockingTest {
+    val job = Job()
+    val channelCallback = object : IStorageChannelCallback {
+      override fun onCreate(channel: IStorageChannel?) {
+        assertThat(channel).isInstanceOf(StorageChannelImpl::class.java)
+        job.complete()
+      }
+      override fun asBinder(): IBinder? { return null }
+    }
+    val messageCallback = object : IMessageCallback {
+      override fun onMessage(encodedMessage: ByteArray?) {}
+
+      override fun asBinder(): IBinder? { return null }
+    }
+
+    val storageServiceNg = createStorageService(this)
+
+    val storeOptions = StoreOptions(storageKey, CountType())
+    val encodedStoreOptions = with(Parcel.obtain()) {
+      writeStoreOptions(storeOptions, ParcelableCrdtType.Count, 0)
+      marshall()
+    }
```

Comment:
A helper extension function would be great here:

```
private fun StoreOptions.toParcel(): Parcel = with(Parcel.obtain()) {
  ...
}
```

Reply:
I've called this helper function: `toParcelToByteArray()` because it writes the StoreOptions to a Parcel and then returns the raw bytes of that parcel.. alternative name suggestions are welcome :)

Reply:
Sounds good! Yep that's the right place to put it.

Other naming suggestions: `toParcelByteArray()` or `toParcelBytes()`, but what you have is fine already.